### PR TITLE
Not equal

### PR DESCRIPTION
**Pull request by ikalinin1**

### CODE REVIEW

File: docs/modules/plugins/pages/plugin-visual.adoc [hunk 1]
```diff
@@ -101,6 +102,10 @@ image::checkpoint.png[Checkpoint view]
 
 b) Contextual click on the checkpoint image -> save as to baseline folder
 
+=== `COMPARE_NOT_EQUAL`
+
+Work the same way as <<_compare_against>> except it checks if actual appearance differs from the baseline.
```
Comment (valfirst) [inline docs/modules/plugins/pages/plugin-visual.adoc:107]: please, add details how will `acceptable-diff-percentage` work in this case

File: vividus-extension-visual-testing/src/main/java/org/vividus/visual/bdd/AbstractVisualSteps.java [hunk 2]
```diff
@@ -54,5 +58,14 @@ public AbstractVisualSteps(IUiContext uiContext, IAttachmentPublisher attachment
...
     }
 
-    protected abstract void verifyResult(VisualCheckResult result);
+    protected void verifyResult(VisualCheckResult result)
+    {
+        boolean passed = result.isPassed() ^ result.getActionType() == VisualActionType.COMPARE_NOT_EQUAL;
+        softAssert.assertTrue("Visual check passed", passed);
+    }
+
+    protected ISoftAssert softAssert()
```
Comment (valfirst) [inline vividus-extension-visual-testing/src/main/java/org/vividus/visual/bdd/AbstractVisualSteps.java:67]:
```suggestion
    protected ISoftAssert getSoftAssert()
```

File: vividus-plugin-visual/src/main/java/org/vividus/visual/VisualSteps.java [hunk 3]
```diff
@@ -95,13 +93,17 @@ private void performVisualAction(Supplier<VisualCheck> visualCheckFactory)
...
-                return check.getAction() == VisualActionType.COMPARE_AGAINST
-                                         ? visualTestingEngine.compareAgainst(check)
-                                         : visualTestingEngine.establish(check);
+                switch (check.getAction())
+                {
+                    case ESTABLISH:
+                        return visualTestingEngine.establish(check);
+                    default:
+                        return visualTestingEngine.compareAgainst(check);
+                }
```
Comment (valfirst) [inline vividus-plugin-visual/src/main/java/org/vividus/visual/VisualSteps.java:102]:
```suggestion
                return check.getAction() == VisualActionType.ESTABLISH
                                         ? visualTestingEngine. establish(check)                                         
                                         : visualTestingEngine.compareAgainst(check);
```

File: vividus-extension-visual-testing/src/main/java/org/vividus/visual/model/VisualActionType.java [hunk 4]
```diff
@@ -19,5 +19,6 @@
 public enum VisualActionType
 {
     ESTABLISH,
-    COMPARE_AGAINST
+    COMPARE_AGAINST,
+    COMPARE_NOT_EQUAL
```
Comment (valfirst) [inline vividus-extension-visual-testing/src/main/java/org/vividus/visual/model/VisualActionType.java:23]:
```suggestion
    CHECK_INEQUALITY_AGAINST
```

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-plugin-visual/src/main/java/org/vividus/visual/VisualSteps.java [hunk 5]
```diff
@@ -135,40 +136,55 @@ public void runVisualTests(VisualActionType actionType, String name, ExamplesTab
...
+            visualCheck.setAcceptableDiffPercentage(getParameter(rowAsParameters,
+                    ACCEPTABLE_DIFF_PERCENTAGE_COLUMN_NAME));
+        }
+    }
+
+    private OptionalInt getParameter(Parameters rowAsParameters, String paramterName)
+    {
+        return rowAsParameters.values().containsKey(paramterName)
+                ? OptionalInt.of(rowAsParameters.valueAs(paramterName, Integer.TYPE))
+                : OptionalInt.empty();
```
Comment (valfirst) [inline vividus-plugin-visual/src/main/java/org/vividus/visual/VisualSteps.java:185]: please fix cover this logic with tests

**Review by valfirst — CHANGES_REQUESTED** [review]

**Review by valfirst — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**